tests/unit/onyxTest.js

```diff
@@ -178,6 +178,42 @@ describe('Onyx', () => {
             });
     });
 
+    it('should overwrite an array key nested inside an object', () => {
+        let testKeyValue;
+        connectionID = Onyx.connect({
+            key: ONYX_KEYS.TEST_KEY,
+            initWithStoredValues: false,
+            callback: (value) => {
+                testKeyValue = value;
+            },
+        });
+
+        Onyx.merge(ONYX_KEYS.TEST_KEY, {something: [1, 2, 3]});
+        Onyx.merge(ONYX_KEYS.TEST_KEY, {something: [4]});
+        return waitForPromisesToResolve()
+            .then(() => {
+                expect(testKeyValue).toEqual({something: [4]});
+            });
+    });
+
+    it('should overwrite an array key nested inside an object when using merge', () => {
```

Comment:
Could you please change this to say that the difference here is that it has to do with collections? I'm pretty confused on how this test is different than the one above it.